src/sentry/identity/oauth2.py

```diff
@@ -355,7 +355,7 @@ def dispatch(self, request: Request, pipeline) -> HttpResponse:
             if error:
                 pipeline.logger.info("identity.token-exchange-error", extra={"error": error})
                 lifecycle.record_failure(
-                    {"failure_reason": "token_exchange_error", "msg": ERR_INVALID_STATE}
+                    "token_exchange_error", extra={"failure_info": ERR_INVALID_STATE}
```

Comment:
This is what broke in production. I don't see any other instances besides the two in the file.